src/app/shared/helpers/state.ts

```diff
@@ -1,8 +1,16 @@
+import { HttpErrorResponse } from '@angular/common/http';
 
 export type FetchState<T> = Ready<T>|Fetching<T>|FetchError;
 export type Ready<T> = Readonly<{ tag: 'ready', isReady: true, data: T, isFetching: false, isError: false, error?: undefined }>;
 export type Fetching<T> = Readonly<{ tag: 'fetching', isReady: false, data?: T, isFetching: true, isError: false, error?: undefined }>;
-export type FetchError = Readonly<{ tag: 'error', isReady: false, data?: undefined, isFetching: false, isError: true, error: Error }>;
+export type FetchError = Readonly<{
+  tag: 'error',
+  isReady: false,
+  data?: undefined,
+  isFetching: false,
+  isError: true,
+  error: Error|HttpErrorResponse,
```

Comment:
You do the same change as Alex a few weeks ago.. let me find my answer...
> Hum I don't like the idea of polluting the error with possible more-specific errors. It's clear there are improvement to be done (for instance, having a fetch state with a type of data and a type of error).
I think the most pragmatic approach would be to keep the previous version of the state.ts file and to use in template $any(state.error).status !== 403 so that state.error is converted to any to try to fetch more specific error properties.

Ok we are not exactly in the same but still, changing "my number is a positive integer" by "my number is a positive integer or a even positive integer" does not not feel right in terms of inheritance.

Reply:
Problem is that HttpErrorResponse does not inherit the built-in Error class…

Reply:
1. `Error` of es5 used here seems to be a interface. (so also the remark on assign vs spread operator looks strange)
2. `class HttpErrorResponse extends HttpResponseBase implements Error `, so that looks like there are not unrelated.

Reply:
fixed